test/jdk/java/awt/TitledBorder/TitledBorderTest.java

```diff
@@ -0,0 +1,266 @@
+                    System.out.println(scaling + " : " + y);
+
+                     checkHorizontalBorder(y, img, scaling);
+                }
+        }
+    }
+
+    private static void hBorderLoop(int x1, int x2, int y, BufferedImage buff,
+                                    Color c, double scaling) throws RuntimeException {
+        int thickness = 0;
+        for (int i = x1; i < x2; i++) {
+            if (buff.getRGB(i, y) == c.getRGB()) thickness++;
+        }
+
+        int expected = (int) Math.floor(scaling);
+        if (thickness > expected) {
+            throw new RuntimeException("Horizontal Border drawn too thick. Thickness: "
+                    + thickness + " Scaling: " + scaling + " y: " + y);
+        } else if (thickness < expected) {
+            throw new RuntimeException("Horizontal Border drawn too thin. Thickness: "
+                    + thickness + " Scaling: " + scaling + " y: " + y);
+        }
+    }
+
+    private static void checkHorizontalBorder(int y, BufferedImage img, double scaling) throws RuntimeException {
+        // checking left border
+        hBorderLoop(0, (int) (Math.floor(scaling)+(5*scaling)), y, img, shadow, scaling);
+        hBorderLoop((int) Math.floor(scaling), (int) (Math.floor(scaling)*2+5*scaling), y, img, highlight, scaling);
+
+        // checking right border
+        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)*2+5*scaling)), img.getWidth() - ((int) Math.floor(scaling)),
+                y, img, shadow, scaling);
+        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)+5*scaling)), img.getWidth(),
+                y, img, highlight, scaling);
+    }
+
+    private static void verifyThickness(int x, int thickness, double scaling) {
+        int expected = (int) Math.floor(scaling);
+        if (thickness < expected) throw new RuntimeException("Vertical Border drawn too thin. Thickness: "
+                + thickness + " Scaling: " + scaling + " x: " + x);
+        if (thickness > expected) throw new RuntimeException("Vertical Border drawn too thick. Thickness: "
+                + thickness + " Scaling: " + scaling + " x: " + x);
+    }
+
+    private static void checkVerticalBorder(int x, BufferedImage img, double scaling) throws RuntimeException {
+        int thickness = 0;
+        boolean checkShadow = false;
+        boolean checkHighlight = false;
+        for (int i = 0; i < img.getHeight(); i++) {
+            int color = img.getRGB(x,i);
+            if (!checkHighlight && !checkShadow) {
+                if (color == shadow.getRGB()) {
+                    checkHighlight = true;
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    throw new RuntimeException("Vertical Border was clipped or overdrawn."
+                            + " Scaling: " + scaling + " x: " + x);
+                } else {
+                    continue;
```

Comment:
`continue` statement is redundant because you can't get into another `if` block.